geode-apis-compatible-with-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/cluster/ClusterSlotsAndNodesDUnitTest.java

```diff
@@ -244,6 +249,42 @@ public void slotsAreNotMissingOrDuplicatedWhenPrimariesAreMoving() throws Except
     assertThat(iterations).isGreaterThan(0);
   }
 
+  @Test
+  public void hostAndPortInfoIsUnique_whenPrimariesAreMoving() throws Exception {
+    AtomicBoolean done = new AtomicBoolean();
+    CompletableFuture<Void> startupShutdownFuture = executor.runAsync(() -> {
+      while (!done.get()) {
+        server2.stop();
+        server2 = cluster.startRedisVM(2, locator.getPort());
+        rebalanceAllRegions(server1);
+      }
+    });
+
+    long endTime = System.currentTimeMillis() + 60_000;
+    CompletableFuture<Integer> getSlotsFuture = executor.supplyAsync(() -> {
+      int iterations = 0;
+
+      while (System.currentTimeMillis() < endTime) {
+        List<ClusterNode> nodes = ClusterNodes.parseClusterNodes(jedis1.clusterNodes()).getNodes();
+
+        if (nodes.size() != 2) {
+          continue;
+        }
+
+        assertThat(nodes.get(0).port).isNotEqualTo(nodes.get(1).port);
+        iterations++;
+      }
+
+      return iterations;
```

Comment:
Could the assertion that `iterations` is greater than 0 be moved to here to simplify things a little? This change would require changing the type of `getSlotsFuture` to `CompletableFuture<Void>` and using `executor.runAsync()`.

Reply:
In practice it would probably be just fine, however in theory it might mess with cleanup. We still want the sequence below to remain the same, however if `getSlotsFuture.get()` ends up throwing an exception (because the assertion on iterations failed) it would also prevent the subsequent calls to `done.set(true)...`.

Reply:
Ah, good call, I hadn't thought of that